.gitlab-ci.yml

```diff
@@ -1,35 +1,73 @@
 stages:
- - build
- - release
+  - build
 
 variables:
   REGISTRY: registry.gitlab.com
   IMAGE: ${REGISTRY}/${CI_PROJECT_NAMESPACE}/${CI_PROJECT_NAME}
 
 before_script:
- # Enable GitLab registry
- - docker login -u gitlab-ci-token -p ${CI_BUILD_TOKEN} ${REGISTRY}
+  # Enable GitLab registry
+  - docker login -u gitlab-ci-token -p ${CI_BUILD_TOKEN} ${REGISTRY}
```

Comment:
It can't be done due to gitlab-ci job runs in parallel on the same hosts and logout for it's user may brake the logins from the other jobs.

Reply:
> may brake the logins from the other jobs.

This is easy to solve, you can log out at another stage, after `build`. This will guarantee that all jobs of the previous stage have been processed.